app/piping/plumber.py

```diff
@@ -0,0 +1,35 @@
+import re
+
+
+class Plumber(object):
+    def __init__(self, context):
```

Comment:
I'm finding this class a little hard to follow, maybe add some comments to explain whats going on.